### PR TITLE
=act #26300 try not to get stopped by death pact before Unregistration is complete

### LINKED ISSUE #26300: TcpConnection logs noisy DeadLetter of TcpConnection$Unregistered
```
Message [akka.io.TcpConnection$Unregistered$] from Actor[akka://default/system/IO-TCP/selectors/$a/0#131005314] to Actor[akka://default/system/IO-TCP/selectors/$a/0#131005314] was not delivered. [1] dead letters encountered. If this is not an expected behavior, then [Actor[akka://default/system/IO-TCP/selectors/$a/0#131005314]]
```

See https://github.com/akka/akka-http/issues/2392

`Unregistered` was recently added. It probably happens because the connection actor signs death pacts with its handler actors. Depending on the order of how things are shut down, the connection actor might already be pulled down by the death pact while waiting for unregistering. We should probably exempt Unregistered from DeadLetter logging.

### PR DESCRIPTION
**Pull request by jrudolph**
Otherwise, there will be noisy DeadLetter logging and also the ordering guarantee we try to hold up, that the connection actor is not closed before the channel is closed, 
is violated.

Fixes #26300.

### CODE REVIEW

File: akka-actor/src/main/scala/akka/io/TcpConnection.scala [hunk 1]
```diff
@@ -359,6 +358,7 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
 
   def stopWith(closeInfo: CloseInformation, shouldAbort: Boolean = false): Unit = {
     closedMessage = Some(closeInfo)
+    unsignDeathPact()
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/io/TcpConnection.scala]: So, the watched actor would stop somewhere between here and that the unregistration completing?

**Review by johanandren — APPROVED** [review]
LGTM if I understood it correct.

File: akka-actor/src/main/scala/akka/io/TcpConnection.scala (hunk 1 above)
Comment (jrudolph) [reply akka-actor/src/main/scala/akka/io/TcpConnection.scala]: That's my guess. The `unregistered` state is so simple that I couldn't come up with any other reason why the actor should be stopped in the meantime.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**